### DISCUSSION TITLE
lockMinter vs setMinter

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#402] [Solidity / smart contracts] [answered]
Question about [lockMinter](https://github.com/nounsDAO/nouns-monorepo/blob/master/packages/nouns-contracts/contracts/NounsToken.sol#L206): When is this intended to be called? Since minter is the auction proxy address, which can change its implementation, what's the difference between setMinter and changing that implementation?

**Comment by solimander** [comment, marked as answer]
We wanted to build as much optionality into the protocol as possible. `setMinter` could be called if a more efficient or otherwise improved proxy implementation comes along, or we decide we don't need a proxy at all. `lockMinter` could be called to ensure full immutability in the Nouns distribution mechanism, if ever desired.
